### DISCUSSION TITLE
Deno package on Linuxbrew

### DISCUSSION THREAD

**Discussion by darkRaspberry** [Homebrew/discussions#1222] [Linux]
As I use brew on ubuntu and it works fantastic. I use most packages like ruby or node etc with brew. But when it comes to deno. It doesn't work on linux. So I created a issue with deno and requested this feature here. https://github.com/denoland/deno_install/issues/180#issuecomment-787837625
But they said deno is maintained with brew. They can't help. I read tons of issue everywhere but didn't reach conculsions.

This is terminal dump clearly shows deno is unavailable in linux.
```shell
dev@ubu27x:~$ brew info deno
deno: stable 1.8.3
Secure runtime for JavaScript and TypeScript
https://deno.land/
Not installed
From: https://github.com/Homebrew/linuxbrew-core/blob/HEAD/Formula/deno.rb
License: MIT
==> Dependencies
Build: llvm ✘, ninja ✘, rust ✘
Required: xz ✘
==> Requirements
Build: Xcode >= 10.0 ✔
Required: macOS ✘
==> Analytics
install: 0 (30 days), 0 (90 days), 9 (365 days)
install-on-request: 0 (30 days), 0 (90 days), 9 (365 days)
build-error: 0 (30 days)

```
I found 4 closed issue [here](https://github.com/Homebrew/linuxbrew-core/issues?q=is%3Aissue+deno+is%3Aclosed) in linux brew formula request.
20 closed PR [here](https://github.com/Homebrew/linuxbrew-core/pulls?q=is%3Apr+deno+is%3Aclosed) in linuxbrew
Some in other places like homebrew repo. 

Giving me a idea about the install script is based on python2 and can't be ported untill it upgraded.

**Comment by SMillerDev** [comment]
> Giving me a idea about the install script is based on python2 and can't be ported untill it upgraded.

Yup, deno requires python version to build that has been unmaintained for well over a year. Once they fix that we can try building on Linux again.

**Reply by darkRaspberry** [reply]
Doesn't deno original maintainers are interested in upgrading their install script.

**Comment by carlocab** [comment]
`deno` itself supports building with Python3. See, for example, https://github.com/denoland/deno/issues/2893, https://github.com/denoland/deno/pull/7779

However, building `deno` currently requires building `v8`, which currently requires Python2.

**Reply by danielnachun** [reply]
I've been able to build recent versions of `v8` with Python 3, so it's possible this is no longer an issue.  I haven't had the chance to dig into this but hope to do so eventually.

**Reply by darkRaspberry** [reply]
🤞🏻

**Reply by SMillerDev** [reply]
It's actually a special version of v8, not the same as the v8 formula.
